pkg/plugin/state.go

```diff
@@ -641,6 +641,11 @@ func (e *AutoscaleEnforcer) reserveResources(
 	}
 
 	shouldDeny := add.VCPU > node.remainingReservableCPU() || add.Mem > node.remainingReservableMem()
+
+	if shouldDeny {
+		e.metrics.ReserveShouldDeny(pod, node)
+	}
```

Comment:
We might want to make this more specific to just `(*AutoscaleEnforcer).Reserve()` (the plugin method), but for now I think it's ok.

Highest priority thing looking at this was making sure that we log something when we increment this metric, which we do.